mongo/datadog_checks/mongo/dbm/operation_samples.py

```diff
@@ -0,0 +1,359 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+
+import time
+from typing import Dict, List, Optional
+
+from bson import json_util
+
+try:
+    import datadog_agent
+except ImportError:
+    from datadog_checks.base.stubs import datadog_agent
+
+from datadog_checks.base.utils.db.sql import compute_exec_plan_signature
+from datadog_checks.base.utils.db.utils import DBMAsyncJob
+from datadog_checks.base.utils.tracking import tracked_method
+from datadog_checks.mongo.common import ReplicaSetDeployment
+
+from .types import (
+    OperationActivityEvent,
+    OperationSampleActivityRecord,
+    OperationSampleClient,
+    OperationSampleEvent,
+    OperationSampleOperationMetadata,
+    OperationSampleOperationStats,
+    OperationSamplePlan,
+)
+
+
+class MongoDbExplainExceptionBase(Exception):
+    pass
+
+
+SAMPLE_EXCLUDE_KEYS = {
+    'statement',
+    'application',
+    'dbname',
+    'user',
+    'client',
+}
+
+
+def agent_check_getter(self):
+    return self._check
+
+
+class MongoOperationSamples(DBMAsyncJob):
+    def __init__(self, check):
+        self._operation_samples_config = check._config.operation_samples
+        self._collection_interval = self._operation_samples_config['collection_interval']
+
+        super(MongoOperationSamples, self).__init__(
+            check,
+            rate_limit=1 / self._collection_interval,
```

Comment:
This defaults to 15, but I think we want 10s for async jobs, no?

Reply:
I switched it to async because every operation sample run usually takes less than 1 second even on a pretty heavy loaded database. We can do more aggressive sampling with async process